scripts/withAppEnvs/buildEnvTypeFile.js

```diff
@@ -0,0 +1,121 @@
+const dotenv = require('dotenv');
+const fs = require('fs');
+const path = require('path');
+
+const rootDir = path.resolve(__dirname, '../..');
+const envDir = path.resolve(rootDir, 'env');
+
+module.exports = function buildEnvTypeFile() {
+    
+  function getPrivateEnvs() {
+    const result = {};
+
+    fs.readdirSync(rootDir)
+      .filter((file) => file.startsWith('.env') && !file.endsWith('.template'))
+      .forEach((file) => {
+        const filePath = path.resolve(rootDir, file);
+        const parsed = dotenv.parse(fs.readFileSync(filePath));  
+
+        Object.entries(parsed).forEach(([key, value]) => {
+          if (key.startsWith('NEXT_PUBLIC')) {
+            // Red color error
+            console.error('\x1b[31m%s\x1b[0m', 
+            `
+❌ NEXT_PUBLIC env vars are not allowed in root .env files.\n 
+
+File: ${file}
+Env: ${key}=${value} \n
+
+- If you're sure this env is not secret and can be shared, please move it to /env/.env.[APP_ENV]
+- If this env is secret and cannot be shared, please remove NEXT_PUBLIC prefix from it
+            `);
+            process.exit(1);
+          }
+          result[key] = value;
+        });
+      });
+
+    return result;
+  }
+
+  function getPublicEnvs() {
```

Comment:
moze wystarczy jedna funkcja `getEnvs(public?:boolean)`? bo na pierwszy rzut oka rozni to sie tylko `!key.startsWith`